pkg/grpc/actions/auth/add_user_to_group.go

```diff
@@ -25,10 +26,40 @@ func AddUserToGroup(ctx context.Context, req *GroupUserRequest, authService auth
 		return err
 	}
 
-	err = authService.AddUserToGroup(req.DomainID, domainType, req.UserID, req.GroupName)
+	// Handle user identification - either by user_id or user_email
+	var userID string
+	if req.UserID != "" {
+		err := actions.ValidateUUIDs(req.UserID)
+		if err != nil {
+			return status.Error(codes.InvalidArgument, "invalid user ID")
+		}
+		userID = req.UserID
```

Comment:
We should still validate that this user ID exists here. Also, same comment about excessive branching making this code hard to read